app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java

```diff
@@ -82,6 +83,12 @@ public void addDescription(Description description) {
         notifyItemInserted(descriptions.size());
     }
 
+    private void removeDescription(int position, Description description) {
+        descriptions.remove(description);
+        notifyItemRemoved(position);
+        notifyItemRangeChanged(position, descriptions.size());
```

Comment:
I don't think it is necessary to say `changed`, I could be misremembering but that is only for content changes in the data

Reply:
I first tried it without this line but it wasn't working as expected! There was some issue when I rotate the device and try to delete it. I got some out of bound error.

Reply:
Then fixing the error is the better choice